### ISSUE TITLE
Cannot use cli tool because it sends login credentials over HTTPS instead of HTTP

### ISSUE THREAD

**Issue by uwagura** [mercator-ocean/copernicus-marine-toolbox#168]
It looks like the copernicusmarine command line tool currently posts login credentials via HTTP instead of HTTPS, based on the output of running `strace` on the `subset` command. When I run 
```
copernicusmarine subset -i cmems_mod_glo_phy-thetao_anfc_0.083deg_PT6H-i -v thetao -t 2022-01-01T00:00:00 -T 2022-12-31T23:59:59 -x -6.17 -X -5.08 -y 35.75 -Y 36.30 -z 0.0 -Z 5.0 --log-level=DEBUG
```
I get the following output (shortened for clarity):
```
DEBUG - 2024-10-15T13:41:27Z - POSTing credentials to https://cmems-cas.cls.fr/cas/login?service=copernicus-marine-client...
< other strace output > 
connect(4, {sa_family=AF_INET, sin_port=htons(80),
```

I am currently working on a system that does not allow POST requests over HTTP, so my connection just hangs before eventually timing out. 

Is there a particular reason why credentials are posted over HTTP instead of HTTPS? It looks like other copernicusmarine commands use HTTPS. And if this is something that cannot be changed, are there any suggested work arounds for using the cli tool? Thanks in advance!

**Comment by renaudjester** [comment]
Hi @uwagura! Thanks for your report!

Indeed it seems that the `login` functionality does some redirections that end up calling an http endpoint that itself leads to https!

<img width="340" alt="Screenshot 2024-10-17 at 09 21 43" src="https://github.com/user-attachments/assets/a24b393a-f9b0-4dcd-848a-1461ee15000f">

The two GET requests with a  "301" response code are http calls!

I don't know why this is like this. I will ask. 

Note that this authentication method will soon be deprecated and that a new login system should be put in place for Copernicus Marine services to replace it. I don't have an exact ETA but the v2.0.0 of the toolbox should support this new system and it should use only https. 

I am not sure to be able to indicate you a work around right now 🤔

**Comment by uwagura** [comment]
@renaudjester , thanks for looking into this. For now, I've been able to get around this by using a proxy to a machine that does allow HTTP post requests. I look forward to the new login system!

**Comment by renaudjester** [comment]
@uwagura Happy you found a solution! Closing this since this should be solved with the v2.0.0 coming (you can already test it in [v2.0.0a4](https://pypi.org/project/copernicusmarine/2.0.0a4/))
